DMCompiler/DM/Builders/DMProcBuilder.cs

```diff
@@ -612,8 +612,10 @@ internal sealed class DMProcBuilder(DMObject dmObject, DMProc proc) {
                 implicitTypeCheck = dmTypes.Value.TypePath;
             } else if (!dmTypes.Value.IsAnything) {
                 // "as anything" performs no check. Other values are unimplemented.
-                DMCompiler.UnimplementedWarning(outputVar.Location,
-                    $"As type {dmTypes} in for loops is unimplemented. No type check will be performed.");
+                byte typeId = (byte) dmTypes.Value.Type;
+                proc.CreateFilteredBaseTypesListEnumerator(typeId, dmTypes.Value.Type);
```

Comment:
This is likely your issue. CreateFilteredBaseTypesEnumerator expects to have a list to pop off the stack, but the `list.EmitPushValue()` is called after this instead of before. While the compiler is keeping track of the stack's expected size, this ends up trying to pop something off an empty stack so the compiler raises that error.

Reply:
I may have interpreted this wrong, but I moved the call to CreateFilteredBaseTypesListEnumerator after EmitPushValue, and I still receive the same error.
```C#
// Depending on the var's type and possibly a given "as [types]", an implicit istype() check is performed
DreamPath? implicitTypeCheck = null;
if (dmTypes == null) {
    // No "as" means the var's type will be used
    implicitTypeCheck = lValue?.Path;
} else if (dmTypes.Value.TypePath != null) {
    // "as /datum" will perform a check for /datum
    implicitTypeCheck = dmTypes.Value.TypePath;
}

list.EmitPushValue(dmObject, proc);
if (implicitTypeCheck != null) {
    if (DMObjectTree.TryGetTypeId(implicitTypeCheck.Value, out var filterTypeId)) {
        // Create an enumerator that will do the implicit istype() for us
        proc.CreateFilteredListEnumerator(filterTypeId, implicitTypeCheck.Value);
    } else {
        DMCompiler.Emit(WarningCode.ItemDoesntExist, outputVar.Location,
            $"Cannot filter enumeration by type {implicitTypeCheck.Value}, it does not exist");
        proc.CreateListEnumerator();
    }
} else if (dmTypes != null && !dmTypes.Value.IsAnything) {
    byte typeId = (byte) dmTypes.Value.Type;
    proc.CreateFilteredBaseTypesListEnumerator(typeId, dmTypes.Value.Type);
} else {
    proc.CreateListEnumerator();
}
```

Reply:
NVM got it to work. was an issue with not having a case for the new annotatedbytecode